src/services/mongoose/users.ts

```diff
@@ -27,7 +31,33 @@ export const UsersRepository = (): IUsersRepository => {
     }
   }
 
+  const findByUsername = async (username: Username): Promise<User | RepositoryError> => {
+    const regexUsername = getUsernameRegex()
+    try {
+      if (!username.match(regexUsername)) {
+        return new CouldNotFindError("Invalid username")
+      }
+
+      const result = await User.findOne({ username: caseInsensitiveRegex(username) })
```

Comment:
do you know what is the performance impact of use this? mongo indexes support this?

Reply:
https://github.com/GaloyMoney/galoy/pull/449#discussion_r691613163